operator/v1/types_ingress.go

```diff
@@ -282,6 +287,178 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the schema for securing an IngressController.
+type SecuritySpec struct {
+	// profile defines the schema for a security profile.
+	Profile SecurityProfileSpec `json:"profile"`
+}
+
+// SecurityProfileSpec defines the schema for a security profile.
+// +union
+type SecurityProfileSpec struct {
+	// type is one of type OldSecurityProfile, IntermediateSecurityProfile,
+	// ModernSecurityProfile or CustomSecurityProfile. OldSecurityProfile,
+	// IntermediateSecurityProfile and ModernSecurityProfile profiles map to
+	// security compatiibility configurations defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// If unset, the IntermediateSecurityProfile type is used.
+	//
+	// +unionDiscriminator
+	// +optional
+	Type SecurityProfileType
```

Comment:
I'm not sure if we _should_, but I want to see if my understanding is correct: is it _possible_ to describe every type in terms of a specific `SecurityProfile`? For example, could the docs for "Old" say:

old is a security profile based on https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility and looks like this:

```yaml
cipers:
- ...
- ...
securityProtocol:
  minimumVersion: 
  maximumersion:
dHParamSize: ...
```

If it is possible, is there a reason to hide the specifics?

Reply:
Yes, it is possible. I'll update the godocs with an example of each security profile.

Reply:
@ironcladlou I'm adding a JSON tag for `Type SecurityProfileType`, but I will not include `omitempty` since this type is a `unionDiscriminator`. Please let me know if that should not be the case.